Cattac/Cattac/GameEngine.swift

```diff
@@ -150,13 +161,22 @@ class GameEngine {
         }
     }
     
+    /// Releases all resources associated with this game, called
+    /// when game has ended.
     func end() {
         for ref in movementWatchers.values {
             ref.removeAllObservers()
         }
     }
     
+    /// Trigger state advancement in game engine.
     private func nextState() {
```

Comment:
Or change the function name to something else, having both `nextState` and `advanceState` is confusing. Maybe call it `triggerStateAdvance` or something.